### PR TITLE
Fix/password validation

### PR DESCRIPTION
**Pull request by YuliiaMay**
1.  create a general user scheme;
2.  add regexp for email and password validation
3. fix password validation ("Password must contain at least 6 characters, one uppercase, one number and one special case character")
4. fix email validation
5. add notification if the checkbox is unchecked (register form)

### CODE REVIEW

File: src/components/common/Forms/ChangePasswordForm/ChangePasswordForm.jsx [hunk 1]
```diff
@@ -119,17 +104,17 @@ export const ChangePasswordForm = () => {
...
-                                        value={values.newPassword}
+                                        type={showPassword ? 'text' : 'password'}
+                                        name='password'
+                                        value={values.password}
                                         placeholder='Enter your password'
                                         onBlur={handleBlur}
-                                        error={errors.newPassword || newPasswordError}
-                                        border={errors.newPassword && touched.newPassword && '1px solid red'}
+                                        error={errors.password || passwordError}
+                                        border={errors.password && touched.password && '1px solid red'}
```
Comment (maxpopovitch) [inline src/components/common/Forms/ChangePasswordForm/ChangePasswordForm.jsx]: Please move this styling to css

File: src/components/common/Forms/RegisterForm/RegisterForm.jsx [hunk 2]
```diff
@@ -189,16 +176,28 @@ export const RegisterForm = () => {
...
-                                    By checking this box, you are creating an account and you agree to the <PolicyLink target='_blank' to='/conditions'>Terms & Conditions</PolicyLink> and <PolicyLink target='_blank' to='/policy'>Privacy Policy</PolicyLink>.
-                                </ConfirmationText>
+                                <label>
+                                    <Checkbox>
+                                        <Field
+                                            name='confirmation'
+                                            type='checkbox'
+                                            onClick={handleToggleCheck}
+                                            value={isChecked}
+                                            style={{display: 'none'}}
```
Comment (maxpopovitch) [inline src/components/common/Forms/RegisterForm/RegisterForm.jsx]: Why `display: none`? Just remove this field from the code :)

File: src/components/common/Forms/RegisterForm/RegisterForm.jsx [hunk 3]
```diff
@@ -189,16 +176,28 @@ export const RegisterForm = () => {
...
+                                        />
+                                        <div>
+                                            {
+                                                isChecked
```
Comment (maxpopovitch) [inline src/components/common/Forms/RegisterForm/RegisterForm.jsx]: `<Icon name={isChecked ? 'checked' : 'unchecked'} size={24} />`

File: src/components/common/Forms/ResetPasswordForm/ResetPasswordForm.jsx [hunk 4]
```diff
@@ -9,26 +12,19 @@ import {
...
 };
 
+
 export const ResetPasswordForm = () => {
-    const resetPassword = useAuthStore(state => state.resetPassword);
+    const { resetPassword } = useAuthStore();
     const navigate = useNavigate();
 
     const handleSubmit = (values, { resetForm }) => {
+        console.log("work");
```
Comment (maxpopovitch) [inline src/components/common/Forms/ResetPasswordForm/ResetPasswordForm.jsx]: Is it still needed?

File: src/services/authAPI.js [hunk 5]
```diff
@@ -13,6 +13,7 @@ axios.defaults.baseURL = 'https://dev-backend-b4vo.onrender.com';
  * body: { name, email, password }
  */
 export const fetchRegister = async (values) => {
+    console.log(values);
```
Comment (maxpopovitch) [inline src/services/authAPI.js]: Is it still needed?

File: src/services/authAPI.js [hunk 6]
```diff
@@ -70,12 +71,13 @@ export const fetchRefresh = async () => {
  * body: { email }
  */
 export const fetchResetPassword = async (values) => {
+    console.log('start');
```
Comment (maxpopovitch) [inline src/services/authAPI.js]: Is it still needed?

File: src/services/authAPI.js [hunk 7]
```diff
@@ -70,12 +71,13 @@ export const fetchRefresh = async () => {
...
     try {
         const response = await axios.post(
             `/api/auth/password-reset`,
             values
         );
-
+        console.log('api res:', response);
```
Comment (maxpopovitch) [inline src/services/authAPI.js]: Is it still needed?

File: src/components/common/Forms/ChangePasswordForm/ChangePasswordForm.jsx (hunk 1 above)
Comment (YuliiaMay) [reply src/components/common/Forms/ChangePasswordForm/ChangePasswordForm.jsx]: Code has been fixed by the comment

File: src/components/common/Forms/RegisterForm/RegisterForm.jsx (hunk 2 above)
Comment (YuliiaMay) [reply src/components/common/Forms/RegisterForm/RegisterForm.jsx]: This field is necessary for Formik lib. 

"<Field /> will automagically hook up inputs to Formik. It uses the name attribute to match up with Formik state. <Field /> will default to an HTML <input /> element."

I get state from this.

File: src/components/common/Forms/RegisterForm/RegisterForm.jsx (hunk 3 above)
Comment (YuliiaMay) [reply src/components/common/Forms/RegisterForm/RegisterForm.jsx]: Code has been fixed by the comment

File: src/components/common/Forms/ResetPasswordForm/ResetPasswordForm.jsx (hunk 4 above)
Comment (YuliiaMay) [reply src/components/common/Forms/ResetPasswordForm/ResetPasswordForm.jsx]: has removed

File: src/services/authAPI.js (hunk 5 above)
Comment (YuliiaMay) [reply src/services/authAPI.js]: has removed

File: src/services/authAPI.js (hunk 6 above)
Comment (YuliiaMay) [reply src/services/authAPI.js]: has removed

File: src/services/authAPI.js (hunk 7 above)
Comment (YuliiaMay) [reply src/services/authAPI.js]: has removed

**Review by maxpopovitch — APPROVED** [review]

**Pull request merged**